webpack/components/surveys/Show.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component, PropTypes } from 'react';
+import { hashHistory} from 'react-router';
+
+import { surveyProps } from '../../prop-types/survey_props';
+import { formProps } from '../../prop-types/form_props';
+import VersionInfo from '../VersionInfo';
+
+
+class SurveyShow extends Component{
+  historyBar(survey) {
+    return (
+      <div className="col-md-3 nopadding no-print">
+        <div className="showpage_sidenav_subtitle">
+          <ul className="list-inline">
+            <li className="subtitle_icon"><span className="fa fa-history" aria-hidden="true"></span></li>
+            <li className="subtitle">History</li>
+          </ul>
+        </div>
+        <VersionInfo versionable={survey} versionableType='survey' />
+      </div>
+    );
+  }
+
+  mainContent(survey, forms) {
+    return (
+      <div className="col-md-9 nopadding maincontent">
+        <div className="action_bar no-print">
+          <button className="btn btn-default" onClick={() => window.print()}>Print</button>
+        </div>
+        <div className="maincontent-details">
+          <h3 className="maincontent-item-name"><strong>Name:</strong> {survey.name} </h3>
+          <p className="maincontent-item-info">Version: {survey.version} - Author: {survey.userId} </p>
+          {forms.map((f,i ) =>
+            <div  key={i} className="basic-c-box panel-default survey-form">
+              <div className="panel-heading">
+                <h3 className="panel-title">{f.name}</h3>
+              </div>
+              <div className="box-content">
+                <ul>
+                  {f.questions.map((q,i) =>
+                    <li key={i}>{q.content}</li>
+                  )}
+                </ul>
+              </div>
+            </div>
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    let {survey, forms} = this.props;
+    if(!survey || !forms){
```

Comment:
should forms be required? On the forms show we allow questions to be empty, i.e. if they make a bucket survey but don't add any forms to it yet then they won't be able to view the details since this will hang I think